File: exercism/go/scrabble-score/scrabble_score.go
package scrabble

import (
	"strings"
	"unicode"
)

var POINT_MAP = map[string]int{
	"aeioulnrst": 1,
	"dg":         2,
	"bcmp":       3,
	"fhvwy":      4,
	"k":          5,
	"jx":         8,
	"qz":         10,
}

// alterntive would be a alphabet-size slice with the values,
// or a switch case
func Score(word string) int {

	score := 0
	for _, c := range strings.ToLower(word) {
		for x, y := range POINT_MAP {

			if strings.Contains(x, string(c)) && unicode.IsLetter(c) {
				//fmt.Print(string(c), x, y, "\n")
				score += y
				break
			}
		}
	}
	return score
}

File: exercism/python/scrabble-score/scrabble_score.py
"""
Scrabble Score
"""

POINT_MAP = {
    "aeioulnrst": 1,
    "dg": 2,
    "bcmp": 3,
    "fhvwy": 4,
    "k": 5,
    "jx": 8,
    "qz": 10,
}


def score(word):
    """
    Given a word, compute the Scrabble score for that word.

    simple solution:
    score = 0
    for c in word.lower():
        for x, y in thisdict.items():
            if c in x and c.isalpha():
              score+=y
              break
    return score
    """

    # char dict for easier calculation
    points = {char: point for chars, point in POINT_MAP.items() for char in chars}

    return sum(points[char] for char in word.lower())
